File: 2461/2461.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        int i = 0, j = 0;
        long result = 0;
        long currWindowSum = 0;
        HashSet<Integer> set = new HashSet<>();

        while (j < nums.length) {
            while (set.contains(nums[j])) {
                currWindowSum -= nums[i];
                set.remove(nums[i]);
                i++;
            }

            currWindowSum += nums[j];
            set.add(nums[j]);

            if (j - i + 1 == k) {
                result = Math.max(result, currWindowSum);

                currWindowSum -= nums[i];
                set.remove(nums[i]);
                i++;
            }

            j++;
        }
        return result;
    }
}

File: 2461/2461.py
class Solution:
    def maximum_sum_of_distinct_subarrays_with_l(self, nums: List[int]) -> int:
        l, r, res = 0, len(nums) - 1, 0
        while l < r:
            res = max(res, min(nums[l], nums[r]) * (r - l))
            if nums[l] < nums[r]:
                l += 1
            else:
                r -= 1
        return res